node/tests/p2p_test.go

```diff
@@ -188,42 +188,51 @@ func TestRestartNodeDiscovery(t *testing.T) {
 		nodes[index] = sw.NewFullNode(nodesConfig...)
 	}
 
-	sub, err := nodes[0].Host.EventBus().Subscribe(&event.EvtPeerConnectednessChanged{})
+	identitySub, err := nodes[0].Host.EventBus().Subscribe(&event.EvtPeerIdentificationCompleted{})
```

Comment:
Don't really get the reasoning behind moving to the Identification event. I know we do those header/p2p tests, but still, I am curious if this change really does anything

Reply:
Good question. As per implementation `event.EvtPeerIdentificationCompleted` is emitting when identification of a remote peer is done and also its' done *after* `event.EvtPeerConnectednessChanged`. So, by changing EvtPeerConnectednessChanged to EvtPeerIdentificationCompleted I want:

1) to be 100% sure that connection between peers works smoothly;
2) to give more time for schedular to process waiting goroutines(I hope it helps to avoid adding time.Sleep in test);
3) Avoid using for loop because EvtPeerConnectednessChanged could be received more the once(with different peer.IDs).